include/misc/sem.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *
+ * @brief public sys_sem APIs.
+ */
+
+#ifndef ZEPHYR_INCLUDE_MISC_SEM_H_
+#define ZEPHYR_INCLUDE_MISC_SEM_H_
+
+#ifdef CONFIG_USERSPACE
+/*
+ * sys_sem exists in user memory working as counter semaphore for
+ * user mode thread. And the init value of sys_sem is 0, the max
+ * value is INT_MAX.
+ */
+
+#include <kernel.h>
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * sys_sem structure
+ */
+struct sys_sem {
+	struct k_futex futex;
+};
+
+/**
+ * @brief Initialize a sys_sem.
+ *
+ * This routine initializes a sys_sem instance, prior to its first use.
+ *
+ * @param sem Address of the sys_sem.
+ *
+ * @return N/A
+ */
+void sys_sem_init(struct sys_sem *sem);
+
+/**
+ * @brief Give a sys_sem.
+ *
+ * This routine gives @a sys_sem. The producer of sys_sem will increase
```

Comment:
Please just use the same documentation as k_sem APIs.
The fact that we use atomic operations / futexes is an implementation detail and shouldn't be discussed here.